test/test_nn.py

```diff
@@ -9289,12 +9289,29 @@ def test_unflatten(self):
         tensor_output = unflatten(tensor_input)
         self.assertEqual(tensor_output.size(), torch.Size([2, 2, 5, 5]))
 
+        # Unflatten Tensor (unflattened_size of list)
+        unflatten = nn.Unflatten(dim=1, unflattened_size=[2, 5, 5])
+        tensor_output = unflatten(tensor_input)
+        self.assertEqual(tensor_output.size(), torch.Size([2, 2, 5, 5]))
+
         # Unflatten NamedTensor
 
         unflatten = nn.Unflatten(dim='features', unflattened_size=(('C', 2), ('H', 5), ('W', 5)))
         named_tensor_input = tensor_input.refine_names('N', 'features')
         named_tensor_output = unflatten(named_tensor_input)
-        self.assertEqual(tensor_output.size(), torch.Size([2, 2, 5, 5]))
+        self.assertEqual(named_tensor_output.size(), torch.Size([2, 2, 5, 5]))
```

Comment:
Same thing if these named tensor-related changes stay: enumerate over an iterable of unflattened_sizes.